templates/style.scss

```diff
@@ -84,21 +84,33 @@ div.rustdoc {
         }
     }
 
-    // this is actual fix for docs.rs navigation and rustdoc sidebar
-    position: fixed;
-    top: $top-navbar-height;
-    bottom: 0;
-    left: 0;
-    right: 0;
-    display: block;
-    overflow-y: auto;
+    position: relative;
+}
+
+html {
+    height: 100%;
 }
 
 body {
     padding: 0;
     margin: 0;
-    // Since top navbar is fixed, we need to add padding to the body content.
-    padding-top: $top-navbar-height;
+    height: 100%;
+    overflow: hidden;
```

Comment:
This one is making me really uncomfortable. What is it used for?

Reply:
This is the entire magic of the change: The "whole page" is divided up like this:

```
┌─ body ───────────────────────────────────────────────────────────────┐
│┌───────────────────────────────────────────────────────────────────┐ │
││                           nav bar                                 │ │
│├───────────────────────────────────────────────────────────────────┤ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                     actual page body with overflow-y: auto        │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
││                                                                   │ │
└┴───────────────────────────────────────────────────────────────────┴─┘
```

* The body does not scroll: Everything inside it remains in place
* The nav bar sits in the top portion of the body, taking up a fixed amount of space, and nothing is "under" (in the z direction) it.
* The main page body is the second, huge, portion of the viewport, and that has `overflow-y: auto` so that it scrolls.

This way, the "meat" of the page remains scrollable, and the nav bar remains where it should be.